### PR TITLE
Adding CI Build

### PR DESCRIPTION
**Pull request by jamiebrynes7**
#### Description
Adding the basic build CI for the FPS starter project - build all three worker assemblies for cloud.
#### Tests
Ran locally.
#### Documentation
N/A - will write up some internal docs.
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.
@jared-improbable

### CODE REVIEW

File: ci/build.sh [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+source .shared-ci/scripts/pinned-tools.sh
+source .shared-ci/scripts/profiling.sh
+
+UNITY_PROJECT_DIR="$(pwd)/workers/unity"
+
+markStartOfBlock "$0"
+
+markStartOfBlock "Building UnityClient"
+
+# TODO - run unity stuff
```
Comment (jared-improbable) [inline ci/build.sh]: Are these TODOs to be resolved in this PR?

Comment (jamiebrynes7) [reply ci/build.sh]: They have already been done 😂

File: ci/build.sh [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+markEndOfBlock "Building UnityGameLogic"
+
+markStartOfBlock "Building SimulatedPlayerCoordinator"
+
+pushd "${UNITY_PROJECT_DIR}"
+    dotnet run -p ../../.shared-ci/tools/RunUnity/RunUnity.csproj -- \
+        -projectPath "${UNITY_PROJECT_DIR}" \
+        -batchmode \
+        -quit \
+        -logfile "$(pwd)/../../logs/UnityGameLogicBuild.log" \
```
Comment (jessicafalk) [inline ci/build.sh]: that looks wrong

File: ci/build.sh [hunk 3]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
```
Comment (jessicafalk) [inline ci/build.sh]: do we also need to build the SimulatedPlayers?

Comment (jamiebrynes7) [reply ci/build.sh]: We do not - they don't have worker assemblies but are started in process by the coordinator (by adding a prefab)

File: ci/bootstrap.sh [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+# Clone the HEAD of the shared CI repo into ".shared-ci"
+
+if [ -d "${SHARED_CI_DIR}" ]; then
+    rm -rf "${SHARED_CI_DIR}"
+fi
+
+git clone "${CLONE_URL}" "${SHARED_CI_DIR}"
+
+
+# Clone the Unity GDK repository
```
Comment (jessicafalk) [inline ci/bootstrap.sh]: SpatialOS GDK for Unity
:trollface:

File: ci/build.sh [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+markStartOfBlock "Building UnityClient"
+
+pushd "${UNITY_PROJECT_DIR}"
+    dotnet run -p ../../.shared-ci/tools/RunUnity/RunUnity.csproj -- \
+        -projectPath "${UNITY_PROJECT_DIR}" \
+        -batchmode \
+        -quit \
+        -logfile "$(pwd)/../../logs/UnityClientBuild.log" \
+        -executeMethod "Improbable.Gdk.BuildSystem.WorkerBuilder.Build" \
+        ${UNITY_ASSET_CACHE_IP:+-CacheServerIPAddress "${UNITY_ASSET_CACHE_IP}"} \
```
Comment (jared-improbable) [inline ci/build.sh]: Extract this out into a variable + comment to explain the weird bash syntax.

**Pull request merged**